sdk-java/src/main/java/ly/count/sdk/java/internal/ModuleCrash.java

```diff
@@ -78,12 +78,22 @@ public Integer getFeature() {
     }
 
     public CrashImplCore onCrash(CtxCore ctx, Throwable t, boolean fatal, String name, Map<String, String> segments, String... logs) {
+        if (t == null) {
+            L.e("Throwable cannot be null");
+            return  null;
+        }
         return onCrash(ctx, new CrashImplCore().addThrowable(t).setFatal(fatal).setName(name).setSegments(segments).setLogs(logs));
     }
 
-    public CrashImplCore onCrash(CtxCore ctx, CrashImplCore crash) {
+    private CrashImplCore onCrash(CtxCore ctx, CrashImplCore crash) {
         long running = started == 0 ? 0 : DeviceCore.dev.nsToMs(System.nanoTime() - started);
         crash.putMetricsCore(ctx, running);
+
+        if (!crash.validateMetrics()){
+            L.w("OS name or App version cannot be null or empty");
```

Comment:
You need to explicitly print out which one of them is not set. It should not be a single combined log